qa/bin/clone_and_test_examples

```diff
@@ -21,11 +21,9 @@ if [ -n "$QA" ] || test_on_branch $CIRCLE_BRANCH; then
   if [ -n "$LANGUAGE" ]; then
     LANGS[0]="$LANGUAGE"
   else
-    LANGS[0]="ruby"
-    LANGS[1]="python"
-    LANGS[2]="php"
-    LANGS[3]="java"
-    LANGS[4]="javascript"
+    # N.B. Only QA-ing languages that have been migrated to using our parser
+    # backend
+    LANGS[0]="java"
```

Comment:
This is fine as-is, but seeing the defaults get deleted made me realize we could simplify this approach a good deal & allow overriding the entire array from the outside the script as well.

```
if [ -z "$LANGS" ]; then
  LANGS=(ruby python php java javascript)
fi

for lang in ${LANGS[*]}; do
```

With that formulation, the CI invocation could be `LANGS=java make qa`, and then become `LANGS="java ruby" make qa` to run both, etc. That way instead of having to change the defaults we can hand-pick the ones we want to run at a given time, which seems preferable.

Reply:
Oo I like that. Updated.

I don't totally understand where the string is being split into an array, but I did some local tesitng and it does seem to work...

Reply:
It's shell word-splitting. Arrays are kinda just some bash fanciness on top of standard word-splitting.